### PR TITLE
DASHB-590: Find method added to AccountService

### PR DESCRIPTION
**Pull request by tolusha**
API Account modified, added ability to find accounts by specific criteria simultaneously:
* userId
* userName
* owner's email
* subscription

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@ApiParam(value = "Max items count", required = false)
                                                      @DefaultValue("-1") @QueryParam("maxItems") int maxItems,
                                                      @ApiParam(value = "Skip count", required = false)
                                                      @QueryParam("skipCount") int skipCount,

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java [hunk 1]
```diff
@@ -1038,6 +1041,66 @@ public void redistributeResources(@ApiParam(value = "Account ID", required = tru
...
+            for (Account account : accounts) {
+                result.add(toDescriptor(account, securityContext));
+            }
+            return result;
+        } catch (NumberFormatException e) {
+            throw new ServerException("Page parameter", e);
+        }
+    }
+
+    private int parseInt(String param, String name) throws ServerException {
```
Comment (skabashnyuk) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java]: you don't need this method

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    String getId();
+
+    void setId(String id);
+
+    String getEmailOwner();
+
+    void setEmailOwner(String emailOwner);
+
+    String getSubscription();
```
Comment (sleshchenko) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: What is subscription? SubscriptionId, ServiceId, PlanId? It isn't clear

File: platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalAccountDaoImpl.java [hunk 3]
```diff
@@ -414,4 +423,55 @@ public SubscriptionQueryBuilder getSubscriptionQueryBuilder() {
...
+                List<Account> bySubscriptions = new LinkedList<>();
+                for (Subscription subscription : subscriptions) {
+                    if (subscription.getServiceId().equals(searchCriteria.getSubscription())) {
+                        bySubscriptions.add(getById(subscription.getAccountId()));
+                    }
+                }
+                merge(result, bySubscriptions);
+            }
+
+        } catch (NotFoundException e) {
```
Comment (skabashnyuk) [inline platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalAccountDaoImpl.java]: Try catch set is too wide. NotFound exception can be in getById(searchCriteria.getId()); and as rusult you will have empty list

Comment (tolusha) [inline platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalAccountDaoImpl.java]: It what I am suppose to have.

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java (hunk 2 above)
Comment (tolusha) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: ServiceId

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java (hunk 1 above)
Comment (tolusha) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/server/AccountService.java]: Which one?

**Comment by tolusha** [comment]
Why do you think that 'skipCount' and 'maxItems' are better than 'page' and 'perPage' ones?

**Comment by skabashnyuk** [comment]
AFAIK some time ago we agree that they are standart for paging in our API
+ you don't need to parse strings

**Comment by tolusha** [comment]
It's not convinient, we display page and some number of items per page.

**Comment by tolusha** [comment]
I need to validate passed parameters anyway, so I prefer to parse them.

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.List;
+
+/**
+ * @author Anatoliy Bazko
+ */
+@DTO
+public interface AccountSearchCriteria {
+
+    List<String> getIds();
```
Comment (skabashnyuk) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: Is this accountId?

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    void setIds(List<String> ids);
+
+    List<String> getOwnerIds();
+
+    void setOwnerIds(List<String> ownerIds);
+
+    String getServiceId();
```
Comment (skabashnyuk) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: Is this SubscriptionPlanId?

Comment (tolusha) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: No, it is serviceId like OnPremises or Saas

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java (hunk 4 above)
Comment (tolusha) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: account id

Comment (skabashnyuk) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: Shall we rename it to accountId to be clear?

File: platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java (hunk 5 above)
Comment (tolusha) [inline platform-api/che-core-api-account/src/main/java/org/eclipse/che/api/account/shared/dto/AccountSearchCriteria.java]: ids -> accountIds changed

**Comment by skabashnyuk** [comment]
will be done as part of https://jira.codenvycorp.com/browse/IDEX-2414

**Pull request closed without merging**